front/src/utils/date.ts

```diff
@@ -218,18 +218,18 @@ export function extractDateAndTimefromISO(arrivalTime: string, dateFormat: strin
 /**
  * Checks if the given arrival date falls within the specified search time window.
  *
- * @param {string} arrivalTime - The arrival time as a string, which will be parsed into a Date object.
+ * @param {string | Date} arrivalTime - The arrival time, which can be a Date object or an ISO string that will be parsed into a Date object.
  * @param {{ begin: Date; end: Date } | undefined} searchDatetimeWindow - An object containing the start and end dates of the search window. If undefined, the function will return true.
  * @returns {boolean} - Returns true if the arrival date is within the search time window, or if the search time window is undefined. Returns false otherwise.
  */
 export function isArrivalDateInSearchTimeWindow(
-  arrivalTime: string,
+  arrivalTime: string | Date,
```

Comment:
I think it would be better to transform this function so it can only takes a date, can you add a commit to do this ?